widgetastic_manageiq/__init__.py

```diff
@@ -1678,6 +1678,150 @@ def read(self):
+    """ represents toolbar's Attribute selector control
+        it is present on pages of Compare Selected items
+
+    .. code-block:: python
+
+        attribute_selector = View.nested(CompareToolBarAttributeSelector)
+
+        attribute_selector.select('Attributes with different values')
+        attribute_selector.selected
+    """
+    ROOT = './/div[contains(@class, "toolbar-pf-actions")]'
+
+    all_values_button = Button(title="All attributes")
+    diff_values_button = Button(title="Attributes with different values")
+    same_values_button = Button(title="Attributes with same values")
+
+    @property
+    def _view_buttons(self):
+        yield self.all_values_button
+        yield self.diff_values_button
+        yield self.same_values_button
+
+    def select(self, title):
+        for button in self._view_buttons:
+            if button.title == title:
+                return button.click()
+        else:
+            raise ValueError("The Attribute with title {title} isn't present".format(title=title))
+
+    @property
+    def selected(self):
+        if self.is_displayed:
+            return next(btn.title for btn in self._view_buttons if btn.active)
+        else:
+            return None
+
+    def read(self):
+        return self.selected
+
+    @property
+    def is_displayed(self):
+        return self.all_values_button.is_displayed
+
+
+class CompareToolBarModeSelector(View):
+    """ represents toolbar's Mode selector control
+        it is present on pages of Compare Selected items
+
+    .. code-block:: python
+
+        mode_selector = View.nested(CompareToolBarModeSelector)
+
+        mode_selector.select('Exists Mode')
+        mode_selector.selected
+    """
+    ROOT = './/div[contains(@class, "toolbar-pf-actions")]'
+
+    details_mode = Button(title="Details Mode")
+    exists_mode = Button(title="Exists Mode")
+
+    @property
+    def _view_buttons(self):
+        yield self.details_mode
+        yield self.exists_mode
+
+    def select(self, title):
+        for button in self._view_buttons:
+            if button.title == title:
+                return button.click()
+        else:
+            raise ValueError("The Mode with title {title} isn't present".format(title=title))
+
+    @property
+    def selected(self):
+        if self.is_displayed:
+            return next(btn.title for btn in self._view_buttons if btn.active)
+        else:
+            return None
+
+    def read(self):
+        return self.selected
+
+    @property
+    def is_displayed(self):
+        return self.details_mode.is_displayed
+
+
+class CompareToolBarViewSelector(View):
+    """ represents toolbar's view selector control
+        it is present on pages of Compare Selected items
+
+    .. code-block:: python
+
+        view_selector = View.nested(CompareToolBarViewSelector)
+
+        view_selector.select('Compressed View')
+        view_selector.selected
+    """
+    ROOT = './/div[contains(@class, "toolbar-pf-actions")]'
```

Comment:
all 3 selectors use the same ROOT whether this is correct ?

Reply:
From my understanding, yes. As they commonly share div tag with class "toolbar-pf-actions". Please correct me, If I am wrong or something to improve as a locator. @izapolsk

Reply:
@digitronik, widgets should have unique ROOT if it is possible for sure.